Sources/Sentry/Profiling/SentryProfilerState.mm

```diff
@@ -71,36 +79,38 @@ - (void)mutate:(void (^)(SentryProfilerMutableState *))block
     block(_mutableState);
 }
 
+- (void)cacheMainThreadID
+{
+    std::lock_guard<std::mutex> l(_lock);
+    NSAssert([NSThread isMainThread], @"Must be called on main thread");
+    const auto currentThread = pthread_mach_thread_np(pthread_self());
+    if (MACH_PORT_VALID(currentThread)) {
+        _mainThreadID = currentThread;
+    }
+}
+
 - (void)appendBacktrace:(const Backtrace &)backtrace
 {
     [self mutate:^(SentryProfilerMutableState *state) {
         const auto threadID = sentry_stringForUInt64(backtrace.threadMetadata.threadID);
 
-        NSString *queueAddress = nil;
-        if (backtrace.queueMetadata.address != 0) {
-            queueAddress = sentry_formatHexAddressUInt64(backtrace.queueMetadata.address);
-        }
         NSMutableDictionary<NSString *, id> *metadata = state.threadMetadata[threadID];
         if (metadata == nil) {
             metadata = [NSMutableDictionary<NSString *, id> dictionary];
             state.threadMetadata[threadID] = metadata;
         }
-        if (!backtrace.threadMetadata.name.empty() && metadata[@"name"] == nil) {
-            metadata[@"name"] =
-                [NSString stringWithUTF8String:backtrace.threadMetadata.name.c_str()];
+        if (metadata[@"name"] == nil) {
```

Comment:
`m`: I think we shoud first check if `backtrace.threadMetadata.threadID == self->_mainThreadID`. Otherwise, the main thread could have different names depending on the platform. Or is this the desired behavior?

Reply:
This looks like it's checking if it's already been set previously, as we reuse the metadata dictionary. The check you mention does happen before we actually go to set the value on line 107 below in this hunk.

Reply:
Yeah @armcknight is right, that was the intent here